manifests/config.pp

```diff
@@ -1,58 +1,68 @@
+# Class: check_mk::config
+#
+# Configures check_mk.
+# 
+# @summary Configures check_mk. Should not be called directly.
+# 
 
 class check_mk::config (
-  $site,
-  $host_groups = undef,
+  String $site,
+  Optional[Hash]  $host_groups     = undef,
```

Comment:
Maybe `Hash $host_groups = {}` is preferable to `Optional`.

Reply:
This will break the code because there is a check if host_groups is set:
`  if $host_groups {`

Reply:
sure.  Other changes would be needed.  Let's leave this to a later round of tidying.